src/Nancy/Routing/Route.cs

```diff
@@ -69,77 +42,67 @@ public Route(string method, string path, Func<NancyContext, bool> condition, Fun
         /// <param name="path">The path that the route is declared for.</param>
-        /// <param name="condition">A condition that needs to be satisfied inorder for the route to be eligible for invocation.</param>
-        /// <param name="syncFunc">The action that should take place when the route is invoked.</param>
-        /// <returns>A Route instance</returns>
-        [Obsolete("Sync routes are deprecated (see LegacyNancyModule)")]
-        public static Route FromSync(string name, string method, string path, Func<NancyContext, bool> condition, Func<dynamic, dynamic> syncFunc)
+        /// <param name="condition">A condition that needs to be satisfied inorder for the route to be eligiable for invocation.</param>
+        /// <param name="action">The action that should take place when the route is invoked.</param>
+        public Route(string method, string path, Func<NancyContext, bool> condition, Func<object, CancellationToken, Task<T>> action)
+            : this(string.Empty, method, path, condition, action)
         {
-            return FromSync(new RouteDescription(name, method, path, condition), syncFunc);
         }
 
         /// <summary>
-        /// Wraps a sync delegate in a delegate that returns a task
+        /// Gets or sets the action that should take place when the route is invoked.
         /// </summary>
-        /// <param name="syncFunc">Sync delegate</param>
-        /// <returns>Task wrapped version</returns>
-        private static Func<dynamic, CancellationToken, Task<dynamic>> Wrap(Func<object, object> syncFunc)
-        {
-            return (parameters, context) =>
-                {
-                    var tcs = new TaskCompletionSource<dynamic>();
-
-                    try
-                    {
-                        var result = syncFunc.Invoke(parameters);
-
-                        tcs.SetResult(result);
-                    }
-                    catch (Exception e)
-                    {
-                        tcs.SetException(e);
-                    }
+        /// <value>A <see cref="Func{T,K}"/> that represents the action of the route.</value>
+        public Func<object, CancellationToken, Task<T>> Action { get; set; }
 
-                    return tcs.Task;
-                };
+        /// <summary>
+        /// Invokes the route with the provided <paramref name="parameters"/>.
+        /// </summary>
+        /// <param name="parameters">A <see cref="DynamicDictionary"/> that contains the parameters that should be passed to the route.</param>
+        /// <param name="cancellationToken">Cancellation token</param>
+        /// <returns>A (hot) task of <see cref="Response"/> instance.</returns>
+        public override Task<object> Invoke(DynamicDictionary parameters, CancellationToken cancellationToken)
+        {
+            return this.Action.Invoke(parameters, cancellationToken).ContinueWith<object>(t => t.Result, TaskContinuationOptions.OnlyOnRanToCompletion);

```

Comment:
I noticed further up you called an action like `action(...)` but here you're using `Action.Invoke(...)` I don't believe it makes much different but, consistency? :)